sdk/core/azure-core/src/main/java/com/azure/core/util/serializer/SerializerAdapter.java

```diff
@@ -194,4 +195,18 @@ default String serializeIterable(Iterable<?> iterable, CollectionFormat format)
      * @throws IOException If an I/O error occurs
      */
     <T> T deserialize(HttpHeaders headers, Type type) throws IOException;
+
+    /**
+     * Deserializes the provided header returned from a REST API to en entity instance declared as the model of the
+     * header.
+     *
+     * @param header The header.
+     * @param type The type that represents the deserialized header.
+     * @param <T> The type of the deserialized header.
+     * @return A new instance of the type that represents the deserialized header.
+     * @throws IOException If an I/O error occurs.
+     */
+    default <T> T deserializeHeader(Header header, Type type) throws IOException {
```

Comment:
minor: do we want to keep both methods to deserialize header and headers? SerializerAdapter API is bloated, maybe if we add new methods, would it make sense to add them on Header(s) class instead?

Reply:
Good point, it should be possible to add this to `Header` instead of `SerializerAdapter` as `Header` can get access to package-private methods on `JacksonAdapter` using the accessor pattern used throughout the SDKs